internal/cli/status.go

```diff
@@ -134,14 +134,6 @@ func (c *StatusCommand) RefreshApplicationStatus() error {
 	// Get our API client
 	client := c.project.Client()
 
-	// We now know that we're going to need a runner for the refresh job.
```

Comment:
This was my primary impetus for moving the runner startup logic. Once we know we're going to need to run a job, we need to do a bunch of work:

- Figure out if we should run locally or remotely (based on flags, the project, and the waypoint infra's current state)
- If local, start the local runner
- If remote, check if VCS is dirty, and if so warn

Basically all the work that currently is in `setupLocalJobSystem`. If we don't move that logic into the `doJob` path, we'll need to do it here, which feels messier to me than making the logs and exec operation funcs call `setupLocalJobSystem`.